modules/apps/transfer/types/v3/packet.go

```diff
@@ -13,3 +34,103 @@ func NewFungibleTokenPacketData(
 		Memo:     memo,
 	}
 }
+
+// ValidateBasic is used for validating the token transfer.
+// NOTE: The addresses formats are not validated as the sender and recipient can have different
+// formats defined by their corresponding chains that are not known to IBC.
+func (ftpd FungibleTokenPacketData) ValidateBasic() error {
+	if strings.TrimSpace(ftpd.Sender) == "" {
+		return errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "sender address cannot be blank")
+	}
+
+	if strings.TrimSpace(ftpd.Receiver) == "" {
+		return errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "receiver address cannot be blank")
+	}
+
+	if len(ftpd.Tokens) == 0 {
+		return errorsmod.Wrap(types.ErrInvalidAmount, "tokens cannot be empty")
+	}
+
+	for _, token := range ftpd.Tokens {
```

Comment:
why are we not checking the trace in our validate basic?

Reply:
we are checking the trace in token.Validate()?